how-tos/regex/README.md

```diff
@@ -0,0 +1,46 @@
+# How to use Regex
+
+### What is Regex?
+
+Regex is short for Regular Expression, and is a tool for pattern matching, and is used in Heck for environment ID matching.
+We will be using [Boost Regex](https://www.boost.org/doc/libs/1_31_0/libs/regex/doc/syntax.html) for this how to, as this is what Heck uses in its LookupID dependency.
+
+## Note
+Using Regex incorrectly will cause some performance issues, try to understand Regex before using it in an actual map.
+Using [this tool](https://regex101.com/) helps with learning Regex a lot faster. But be aware that you should still be comfortable with Regex before its used in an actual map.
+
+With that out of the way, here is all the important stuff:
+
+Characters: <br>
+`.` - any character <br>
+`[a-z]` - single character from a to z <br>
+`[a-z0-9A-Z]` - single character from a to z, 0 to 9 and A to Z <br>
+`[:blank:]` - any blank character (space and tab) <br>
+
+
+Amount modifiers: <br>
+`*` - 0 or more characters <br>
+`+` - 1 or more characters  <br>
+`{1,3}` - 1 to 3 characters <br>
+
+OR:
+`(a|b)` - matches both "a" and "b"
+
+Examples: <br>
+`[0-9]+,[0-9]+` - one or more digits followed by a comma followed by one or more digits (412,66 would match, 5a,2 wouldn't) <br>
```

Comment:
("412,66" would match, "5a,2" would not)

Reply:
36c1df5